### PR TITLE
Fix regex for sanitizer SUMMARY error type detection

### LINKED ISSUE #123: odr-violation parsing error
Hello! 
When using the CASR I ran into the problem that a report based on the sanitizer output isn't generated, if the error class is odr-violation. Instead the report is generated based on the casr-gdb. The reason seems to be that regex being used for the SUMMARY string parsing, which also takes ":" into account:
libcasr/src/asan.rs:176:
```
SUMMARY: *(AddressSanitizer|libFuzzer): (\S+)
```
In this case, if the line "odr-violation:" specified in the SUMMARY, then the check in  libcasr/src/execution_class.rs:139 fails. Possible solution: 
```
SUMMARY: *(AddressSanitizer|libFuzzer): ([\w \-\(\)\_]+)
```
Or check for string matches not entirely. 

Attached the screenshot with a part of the problematic SUMMARY-line
casr-error

casr_execution_class

casr_regex_asanrs

### PR DESCRIPTION
**Pull request by SweetVishnya**
Name may end with ':'
For instance, SUMMARY: AddressSanitizer: odr-violation:

Closes #123

### CODE REVIEW

**Review by anfedotoff — APPROVED** [review]

File: libcasr/src/asan.rs [hunk 1]
```diff
@@ -173,7 +173,9 @@ impl Severity for AsanContext {
         if asan_report[0].contains("LeakSanitizer") {
             ExecutionClass::find("memory-leaks")
         } else {
-            let summary = Regex::new(r"SUMMARY: *(AddressSanitizer|libFuzzer): (\S+)").unwrap();
+            let summary =
+                Regex::new(r"SUMMARY: *(AddressSanitizer|libFuzzer): ([0-9A-Za-z_\-\(\)]+)")
```
Comment (hkctkuy) [inline libcasr/src/asan.rs:177]: Are digits needed here?

File: libcasr/src/ubsan.rs [hunk 2]
```diff
@@ -53,7 +53,7 @@ impl Severity for UbsanWarning {
         let description = cap.get(1).unwrap().as_str().to_string();
         // Get short description (from last line)
         let short_description = message.last().unwrap();
-        let re = Regex::new(r"SUMMARY: UndefinedBehaviorSanitizer: (\S+)").unwrap();
+        let re = Regex::new(r"SUMMARY: UndefinedBehaviorSanitizer: ([0-9A-Za-z_\-\(\)]+)").unwrap();
```
Comment (hkctkuy) [inline libcasr/src/ubsan.rs:56]: Same question

File: libcasr/src/asan.rs (hunk 1 above)
Comment (SweetVishnya) [reply libcasr/src/asan.rs:177]: I am not sure. Can there be any UBSAN warnings containing digits? Or can there be something like funcnamewithdigits5-param-overlap?

Comment (SweetVishnya) [reply libcasr/src/asan.rs:177]: Maybe, also, underscope is redundant...

Comment (hkctkuy) [reply libcasr/src/asan.rs:177]: Underscope is not redundant, cuz we have `bad-malloc_usable_size` class

Comment (hkctkuy) [reply libcasr/src/asan.rs:177]: As I know, func names and ubsan errors are not relevant here. This code only parse asan execution_class, I guess

Comment (SweetVishnya) [reply libcasr/src/asan.rs:177]: > As I know, func names and ubsan errors are not relevant here. This code only parse asan execution_class, I guess

However, there is the same regex for UBSAN in this PR.

Comment (hkctkuy) [reply libcasr/src/asan.rs:177]: Ubsan error types don't contain digits too
https://github.com/llvm/llvm-project/blob/main/compiler-rt/lib/ubsan/ubsan_checks.inc

Comment (SweetVishnya) [reply libcasr/src/asan.rs]: removed digits

**Comment by sunforgb** [comment]
image
I think this regex won't parse execution class for libfuzzer "fuzz target exited" : 
https://github.com/ispras/casr/blob/f228b4390a06f8f46a8e7636ade18a08b778f871/libcasr/src/execution_class.rs#L112C36-L112C36

**Comment by SweetVishnya** [comment]
> image I think this regex won't parse execution class for libfuzzer "fuzz target exited" : https://github.com/ispras/casr/blob/f228b4390a06f8f46a8e7636ade18a08b778f871/libcasr/src/execution_class.rs#L112C36-L112C36

Yeah, there is a hack for it https://github.com/ispras/casr/pull/125#issuecomment-1696970053

**Pull request merged**